### ISSUE TITLE
TagFilter created from BaseFilterSet Meta.fields is not using `slug`

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#3154]
### Environment
* Nautobot version (Docker tag too if applicable): 1.5-2.0a
* Python version: 3.7-3.10
* Database platform, version: postgres
* Middleware(s):

### Steps to Reproduce
1. Remove `tag = TagFilter()` from a filterset and add `tags` to the `Meta.fields`
2. Try to filter on tag slugs
3. [Example on next demo](https://next.demo.nautobot.com/dcim/devices/?tags=test)

### Expected Behavior

Filter on tag slug

### Observed Behavior

Exception:
```py
Environment:


Request Method: GET
Request URL: http://localhost:8080/dcim/device-types/?&console_ports=&console_server_ports=&power_ports=&power_outlets=&interfaces=&pass_through_ports=&tags=amber&tags=aqua&q=

Django Version: 3.2.16
Python Version: 3.10.9
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'cacheops',
 'corsheaders',
 'django_filters',
 'django_jinja',
 'django_tables2',
 'django_prometheus',
 'social_django',
 'taggit',
 'timezone_field',
 'nautobot.core.apps.NautobotConstanceConfig',
 'nautobot.core',
 'django.contrib.admin',
 'django_celery_beat',
 'django_celery_results',
 'rest_framework',
 'db_file_storage',
 'nautobot.circuits',
 'nautobot.dcim',
 'nautobot.ipam',
 'nautobot.extras',
 'nautobot.tenancy',
 'nautobot.users',
 'nautobot.utilities',
 'nautobot.virtualization',
 'drf_spectacular',
 'drf_spectacular_sidecar',
 'graphene_django',
 'health_check',
 'health_check.storage',
 'django_extensions',
 'constance.backends.database',
 'django_ajax_tables',
 'debug_toolbar',
 'example_plugin.ExamplePluginConfig',
 'nautobot.extras.tests.example_plugin_dependency']
Installed Middleware:
['debug_toolbar.middleware.DebugToolbarMiddleware',
 'django_prometheus.middleware.PrometheusBeforeMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'nautobot.core.middleware.ExceptionHandlingMiddleware',
 'nautobot.core.middleware.RemoteUserMiddleware',
 'nautobot.core.middleware.ExternalAuthMiddleware',
 'nautobot.core.middleware.ObjectChangeMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware',
 'example_plugin.middleware.ExampleMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/db/models/fields/__init__.py", line 2434, in to_python
    return uuid.UUID(**{input_form: value})
  File "/usr/local/lib/python3.10/uuid.py", line 177, in __init__
    raise ValueError('badly formed hexadecimal UUID string')

During handling of the above exception (badly formed hexadecimal UUID string), another exception occurred:
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/source/nautobot/utilities/views.py", line 114, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/source/nautobot/core/views/generic.py", line 249, in get
    self.queryset = filterset.qs
  File "/usr/local/lib/python3.10/site-packages/django_filters/filterset.py", line 243, in qs
    qs = self.filter_queryset(qs)
  File "/usr/local/lib/python3.10/site-packages/django_filters/filterset.py", line 230, in filter_queryset
    queryset = self.filters[name].filter(queryset, value)
  File "/usr/local/lib/python3.10/site-packages/django_filters/filters.py", line 243, in filter
    qs = self.get_method(qs)(**predicate)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 941, in filter
    return self._filter_or_exclude(False, args, kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 961, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 968, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1416, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1435, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1370, in build_filter
    condition = self.build_lookup(lookups, col, value)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1216, in build_lookup
    lookup = lookup_class(lhs, rhs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/lookups.py", line 25, in __init__
    self.rhs = self.get_prep_lookup()
  File "/usr/local/lib/python3.10/site-packages/django/db/models/fields/related_lookups.py", line 117, in get_prep_lookup
    self.rhs = target_field.get_prep_value(self.rhs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/fields/__init__.py", line 2418, in get_prep_value
    return self.to_python(value)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/fields/__init__.py", line 2436, in to_python
    raise exceptions.ValidationError(

Exception Type: ValidationError at /dcim/device-types/
Exception Value: ['“aqua” is not a valid UUID.']
```

The filter predicate is not being set correctly when the filter is created automatically with `Meta.fields`:

```py
# with tag=TagFilter()
>>> f=DeviceTypeFilterSet().filters.get("tag")
>>> f.get_filter_predicate("aqua")
{'tags__slug': 'aqua'}

# with Meta.fields = ["tags"]
>>> f=DeviceTypeFilterSet().filters.get("tags")
>>> f.get_filter_predicate("aqua")
{'tags': 'aqua'}
```

**Comment by glennmatthews** [comment]
That's weird - how are our existing tag filter tests not catching this?

**Comment by gsnider2195** [comment]
> That's weird - how are our existing tag filter tests not catching this?

The only place I'm seeing tag filter tests is `virtualization` and those filters haven't been converted to use `Meta.fields`

**Comment by gsnider2195** [comment]
It would be a good idea to add a generic tag filter test as part of this issue

**Comment by glennmatthews** [comment]
> > That's weird - how are our existing tag filter tests not catching this?
> 
> The only place I'm seeing tag filter tests is `virtualization` and those filters haven't been converted to use `Meta.fields`

Wow, you're right. I could have *sworn* we had test coverage for this. Definitely agree we should add a test case mixin for this.

**Comment by glennmatthews** [comment]
Resolved for 2.0 by #3456.
